pkg/api/handlers/compat/images.go

```diff
@@ -395,9 +400,21 @@ loop: // break out of for/select infinite loop
 					writeStatusCode(http.StatusInternalServerError)
 				}
 			}
-			if err := enc.Encode(report); err != nil {
-				logrus.Warnf("Failed to json encode error %q", err.Error())
+
+			if err != nil && !progressSent {
+				msg := errorhandling.Cause(err).Error()
+				message := jsonmessage.JSONError{
+					Message: msg,
+				}
+				if err := enc.Encode(message); err != nil {
```

Comment:
We can do `enc.Encode(report.Error)` directly I think.

Reply:
If I encode the message with `enc.Encode(report.Error)` Podman will return this message:
```json
{
   "message" : "initializing source docker://quay.io/idonotexist/idonotexist:dummy: reading manifest dummy in quay.io/idonotexist/idonotexist: unauthorized: access to the requested resource is not authorized"
}
```

Have more details that the return message unwrapped, that is like this:
```json
{
   "message" : "unauthorized: access to the requested resource is not authorized"
}
```
Docker returns the same message as the second option. What is your opinion on this? What do you prefer?

Reply:
Nice catch!  I do not feel strongly about it.  However, some libraries try to string-match errors, so returning the same error as Docker is probably the right thing do to.